src/main/java/org/dataloader/BatchObserver.java

```diff
@@ -0,0 +1,33 @@
+package org.dataloader;
+
+/**
+ * A interface intended as a delegate for other Observer-like classes used in other libraries, to be invoked by the calling
+ * {@link ObserverBatchLoader}.
+ * <p>
+ * Some examples include:
+ * <ul>
+ *     <li>Project Reactor's <a href="https://www.reactive-streams.org/reactive-streams-1.0.4-javadoc/org/reactivestreams/Subscriber.html">{@code Subscriber}</a>
+ *     <li>gRPC's <a href="https://grpc.github.io/grpc-java/javadoc/io/grpc/stub/StreamObserver.html">{@code StreamObserver}</a>
+ *     <li>RX Java's <a href="https://reactivex.io/RxJava/3.x/javadoc/io/reactivex/rxjava3/core/Flowable.html">{@code Flowable}</a>
+ * </ul>
+ * @param <V> the value type of the {@link ObserverBatchLoader}
+ */
+public interface BatchObserver<V> {
+
+    /**
+     * To be called by the {@link ObserverBatchLoader} to load a new value.
+     */
+    void onNext(V value);
+
+    /**
+     * To be called by the {@link ObserverBatchLoader} to indicate all values have been successfully processed.
+     * This {@link BatchObserver} should not have any method invoked after this is called.
+     */
+    void onCompleted();
+
+    /**
+     * To be called by the {@link ObserverBatchLoader} to indicate an unrecoverable error has been encountered.
+     * This {@link BatchObserver} should not have any method invoked after this is called.
+     */
+    void onError(Throwable e);
+}
```

Comment:
This is almost exactly `java.util.concurrent.Flow.Subscriber` but without the subscription part.  I guess one can say that the data loader code IS the subscriber

But I think if this was backed under the covers but reactive code publishing "new items" then something has to subscribe to them - should we expose that and use a Java native class?

Maybe not - because we have other variations of this "observer"

Reply:
ps if we did use `Flow` we would have to move to Java 11 - right now data loader is still java 8 - not a big deal but still

Reply:
All the other graphql-java libs are on Java 11, is there anything blocking dataloader getting upgraded to Java 11 too?

Reply:
No blocking reason to go to 11 - just that we havent

Reply:
If it's easier to do a Java 11 upgrade ahead of this PR I can help out